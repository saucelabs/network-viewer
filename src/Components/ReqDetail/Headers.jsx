import React from 'react';
import PropTypes from 'prop-types';

import General from './headers/General';
import Response from './headers/Response';
import Request from './headers/Request';
import QueryString from './headers/QueryString';
import FormData from './headers/FormData';
import HeaderInfo from './headers/HeaderInfo';
import Styles from './Headers.styles.scss';
import RequestPayload from './headers/RequestPayload';

const Headers = ({ data }) => (!data ? null : (
  <section className={Styles['headers-container']}>
    <HeaderInfo
      component={General}
      data={data}
      eventKey="general"
    />
    <HeaderInfo
      component={Response}
      data={data}
      eventKey="response"
    />
    <HeaderInfo
      component={Request}
      data={data}
      eventKey="request"
    />
    {(data.headers.queryString && data.headers.queryString.length) ? (
      <HeaderInfo
        component={QueryString}
        data={data}
        eventKey="queryString"
        isEncodeEnabled
      />
    ) : null}
    {data.headers.postData && data.headers.postData.params && (
      <HeaderInfo
        component={FormData}
        data={data}
        eventKey="formData"
        isEncodeEnabled
      />
    )}
    {data.headers.postData && data.headers.postData.text && (
      <HeaderInfo
        component={RequestPayload}
        data={data}
        eventKey="requestPayload"
        isParseEnabled
      />
    )}
  </section>
));

Headers.propTypes = {
  data: PropTypes.object,
};

Headers.defaultProps = {
  data: null,
};

export default Headers;
