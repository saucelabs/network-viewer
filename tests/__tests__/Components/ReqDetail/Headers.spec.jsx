import React from 'react';
import { mount } from 'enzyme';

import Headers from './../../../../src/Components/ReqDetail/Headers';

describe('Headers', () => {
  const props = {
    data: {
      domain: 'developer.mozilla.org',
      filename: 'slice',
      index: 0,
      method: 'GET',
      size: 135.04,
      startedDateTime: '2019-10-14T08:47:11.592Z',
      status: 200,
      serverIPAddress: ':80',
      timings: {
        _blocked_queueing: 4.382999992230907,
        blocked: 5.496999992230907,
        connect: -1,
        dns: -1,
        receive: 3.252000024076551,
        send: 0.2819999999999998,
        ssl: -1,
        startTime: 0,
        wait: 26.03899999633059,
      },
      type: 'html',
      url: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice',
      headers: {
        response: [{ name: 'Connection', value: 'keep-alive' }],
        request: [{ name: 'Connection', value: 'keep-alive' }],
      },
    },
    maxTime: 5000,
  };

  it('renders without crashing', () => {
    mount(
      <Headers {...props} />,
    );
  });
});
