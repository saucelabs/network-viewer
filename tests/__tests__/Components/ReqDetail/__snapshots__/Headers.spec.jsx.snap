// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers renders without crashing 1`] = `
<Headers
  data={
    Object {
      "domain": "developer.mozilla.org",
      "filename": "slice",
      "headers": Object {
        "postData": Object {
          "text": "{\\"hello\\": \\"foo\\"}",
        },
        "queryString": Array [],
        "request": Array [
          Object {
            "name": "Connection",
            "value": "keep-alive",
          },
        ],
        "response": Array [
          Object {
            "name": "Connection",
            "value": "keep-alive",
          },
        ],
      },
      "index": 0,
      "method": "GET",
      "serverIPAddress": ":80",
      "size": 135.04,
      "startedDateTime": "2019-10-14T08:47:11.592Z",
      "status": 200,
      "timings": Object {
        "_blocked_queueing": 4.382999992230907,
        "blocked": 5.496999992230907,
        "connect": -1,
        "dns": -1,
        "receive": 3.252000024076551,
        "send": 0.2819999999999998,
        "ssl": -1,
        "startTime": 0,
        "wait": 26.03899999633059,
      },
      "type": "html",
      "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
    }
  }
  maxTime={5000}
>
  <section>
    <HeaderInfo
      component={[Function]}
      data={
        Object {
          "domain": "developer.mozilla.org",
          "filename": "slice",
          "headers": Object {
            "postData": Object {
              "text": "{\\"hello\\": \\"foo\\"}",
            },
            "queryString": Array [],
            "request": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
            "response": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
          },
          "index": 0,
          "method": "GET",
          "serverIPAddress": ":80",
          "size": 135.04,
          "startedDateTime": "2019-10-14T08:47:11.592Z",
          "status": 200,
          "timings": Object {
            "_blocked_queueing": 4.382999992230907,
            "blocked": 5.496999992230907,
            "connect": -1,
            "dns": -1,
            "receive": 3.252000024076551,
            "send": 0.2819999999999998,
            "ssl": -1,
            "startTime": 0,
            "wait": 26.03899999633059,
          },
          "type": "html",
          "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
        }
      }
      eventKey="general"
      isEncodeEnabled={false}
      isParseEnabled={false}
      isVisible={true}
    >
      <div
        className="header-info active"
      >
        <HeaderTitle
          eventKey="general"
          isEncodeEnabled={false}
          isParseEnabled={false}
          isPayloadTransformed={true}
          onClick={[Function]}
          onPayloadTransform={[Function]}
        >
          <div>
            <span
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <IconCaretUp
                className=""
              >
                <svg
                  className=""
                  height="8"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 13 8"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M11.547,8.008 C11.547,8.008 6.500,2.919 6.500,2.919 C6.500,2.919 1.453,8.008 1.453,8.008 C1.453,8.008 0.011,6.554 0.011,6.554 C0.011,6.554 6.500,0.011 6.500,0.011 C6.500,0.011 12.989,6.554 12.989,6.554 C12.989,6.554 11.547,8.008 11.547,8.008 Z"
                      fillRule="evenodd"
                    />
                  </g>
                </svg>
              </IconCaretUp>
              General
            </span>
          </div>
        </HeaderTitle>
        <ChildComponent>
          <div>
            <p
              key="url"
            >
              <span>
                Request URL:
              </span>
              <span>
                https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
              </span>
            </p>
            <p
              key="method"
            >
              <span>
                Request Method:
              </span>
              <span>
                GET
              </span>
            </p>
            <p
              key="status"
            >
              <span>
                Status Code:
              </span>
              <span>
                200
              </span>
            </p>
            <p
              key="serverIPAddress"
            >
              <span>
                Remote Address:
              </span>
              <span>
                :80
              </span>
            </p>
          </div>
        </ChildComponent>
      </div>
    </HeaderInfo>
    <HeaderInfo
      component={[Function]}
      data={
        Object {
          "domain": "developer.mozilla.org",
          "filename": "slice",
          "headers": Object {
            "postData": Object {
              "text": "{\\"hello\\": \\"foo\\"}",
            },
            "queryString": Array [],
            "request": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
            "response": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
          },
          "index": 0,
          "method": "GET",
          "serverIPAddress": ":80",
          "size": 135.04,
          "startedDateTime": "2019-10-14T08:47:11.592Z",
          "status": 200,
          "timings": Object {
            "_blocked_queueing": 4.382999992230907,
            "blocked": 5.496999992230907,
            "connect": -1,
            "dns": -1,
            "receive": 3.252000024076551,
            "send": 0.2819999999999998,
            "ssl": -1,
            "startTime": 0,
            "wait": 26.03899999633059,
          },
          "type": "html",
          "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
        }
      }
      eventKey="request"
      isEncodeEnabled={false}
      isParseEnabled={false}
      isVisible={false}
    >
      <div
        className="header-info"
      >
        <HeaderTitle
          eventKey="request"
          isEncodeEnabled={false}
          isParseEnabled={false}
          isPayloadTransformed={true}
          onClick={[Function]}
          onPayloadTransform={[Function]}
        >
          <div>
            <span
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <IconCaretUp
                className=""
              >
                <svg
                  className=""
                  height="8"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 13 8"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M11.547,8.008 C11.547,8.008 6.500,2.919 6.500,2.919 C6.500,2.919 1.453,8.008 1.453,8.008 C1.453,8.008 0.011,6.554 0.011,6.554 C0.011,6.554 6.500,0.011 6.500,0.011 C6.500,0.011 12.989,6.554 12.989,6.554 C12.989,6.554 11.547,8.008 11.547,8.008 Z"
                      fillRule="evenodd"
                    />
                  </g>
                </svg>
              </IconCaretUp>
              Request Headers
            </span>
          </div>
        </HeaderTitle>
      </div>
    </HeaderInfo>
    <HeaderInfo
      component={[Function]}
      data={
        Object {
          "domain": "developer.mozilla.org",
          "filename": "slice",
          "headers": Object {
            "postData": Object {
              "text": "{\\"hello\\": \\"foo\\"}",
            },
            "queryString": Array [],
            "request": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
            "response": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
          },
          "index": 0,
          "method": "GET",
          "serverIPAddress": ":80",
          "size": 135.04,
          "startedDateTime": "2019-10-14T08:47:11.592Z",
          "status": 200,
          "timings": Object {
            "_blocked_queueing": 4.382999992230907,
            "blocked": 5.496999992230907,
            "connect": -1,
            "dns": -1,
            "receive": 3.252000024076551,
            "send": 0.2819999999999998,
            "ssl": -1,
            "startTime": 0,
            "wait": 26.03899999633059,
          },
          "type": "html",
          "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
        }
      }
      eventKey="response"
      isEncodeEnabled={false}
      isParseEnabled={false}
      isVisible={false}
    >
      <div
        className="header-info"
      >
        <HeaderTitle
          eventKey="response"
          isEncodeEnabled={false}
          isParseEnabled={false}
          isPayloadTransformed={true}
          onClick={[Function]}
          onPayloadTransform={[Function]}
        >
          <div>
            <span
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <IconCaretUp
                className=""
              >
                <svg
                  className=""
                  height="8"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 13 8"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M11.547,8.008 C11.547,8.008 6.500,2.919 6.500,2.919 C6.500,2.919 1.453,8.008 1.453,8.008 C1.453,8.008 0.011,6.554 0.011,6.554 C0.011,6.554 6.500,0.011 6.500,0.011 C6.500,0.011 12.989,6.554 12.989,6.554 C12.989,6.554 11.547,8.008 11.547,8.008 Z"
                      fillRule="evenodd"
                    />
                  </g>
                </svg>
              </IconCaretUp>
              Response Headers
            </span>
          </div>
        </HeaderTitle>
      </div>
    </HeaderInfo>
    <HeaderInfo
      component={[Function]}
      data={
        Object {
          "domain": "developer.mozilla.org",
          "filename": "slice",
          "headers": Object {
            "postData": Object {
              "text": "{\\"hello\\": \\"foo\\"}",
            },
            "queryString": Array [],
            "request": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
            "response": Array [
              Object {
                "name": "Connection",
                "value": "keep-alive",
              },
            ],
          },
          "index": 0,
          "method": "GET",
          "serverIPAddress": ":80",
          "size": 135.04,
          "startedDateTime": "2019-10-14T08:47:11.592Z",
          "status": 200,
          "timings": Object {
            "_blocked_queueing": 4.382999992230907,
            "blocked": 5.496999992230907,
            "connect": -1,
            "dns": -1,
            "receive": 3.252000024076551,
            "send": 0.2819999999999998,
            "ssl": -1,
            "startTime": 0,
            "wait": 26.03899999633059,
          },
          "type": "html",
          "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
        }
      }
      eventKey="requestPayload"
      isEncodeEnabled={false}
      isParseEnabled={true}
      isVisible={false}
    >
      <div
        className="header-info"
      >
        <HeaderTitle
          eventKey="requestPayload"
          isEncodeEnabled={false}
          isParseEnabled={true}
          isPayloadTransformed={true}
          onClick={[Function]}
          onPayloadTransform={[Function]}
        >
          <div>
            <span
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <IconCaretUp
                className=""
              >
                <svg
                  className=""
                  height="8"
                  preserveAspectRatio="xMidYMid"
                  viewBox="0 0 13 8"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M11.547,8.008 C11.547,8.008 6.500,2.919 6.500,2.919 C6.500,2.919 1.453,8.008 1.453,8.008 C1.453,8.008 0.011,6.554 0.011,6.554 C0.011,6.554 6.500,0.011 6.500,0.011 C6.500,0.011 12.989,6.554 12.989,6.554 C12.989,6.554 11.547,8.008 11.547,8.008 Z"
                      fillRule="evenodd"
                    />
                  </g>
                </svg>
              </IconCaretUp>
              Request Payload
            </span>
            <span
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              view source
            </span>
          </div>
        </HeaderTitle>
      </div>
    </HeaderInfo>
  </section>
</Headers>
`;
